### DISCUSSION TITLE
Implement "Ungroup by" transformation for convert multi-valued table to single-valued Time Line

### DISCUSSION THREAD

**Discussion by MurzNN** [grafana/grafana#38302] [Dashboards, Panels & Visualizations]
**What would you like to be added**:
For visualizing tabled data as graph very necessary some sort of "Ungroup by" transformation, that converts multi-valued table like this:
time|Alice|Bob|John|Tom
:-----:|:-----:|:-----:|:-----:|:-----:
2021-01-01 01:01:00|123|456| | 
2021-01-01 01:02:00| |23|12| 
2021-01-01 01:03:00| | |52| 
2021-01-01 01:04:00|123|53| |22

to single-valued TimeLine (for Time Line Graph):
time|label|value
:-----:|:-----:|:-----:
2021-01-01 01:01:00|Alice|123
2021-01-01 01:01:00|Bob|456
2021-01-01 01:02:00|Bob|23
2021-01-01 01:03:00|John|12
2021-01-01 01:03:00|John|52
2021-01-01 01:04:00|Bob|53
2021-01-01 01:04:00|Tom|22
2021-01-01 01:04:00|Alice|123

**Why is this needed**:
This is needed especially for JSON-formatted Loki data source, because logs very often contain list of key-value data with dynamic keys (because variety of people names like in example is unlimited), so workaround like creating separate query for each key is not suitable, and there is no ways to visualize such type of data in Grafana Graph.

**Comment by MurzNN** [comment]
Same problem often happens with SQL query results, when we can receive data from several columns as one query, instead of creating separate query for each column. And we can display them as table, but not as Timeline Graph.

Example of those situations we can see even in Grafana examples here:
- https://grafana.com/docs/grafana/latest/panels/transformations/types-options/#value-field-name
- https://grafana.com/docs/grafana/latest/panels/transformations/types-options/#join-by-field-outer-join
![img](https://grafana.com/static/img/docs/transformations/join-fields-after-7-0.png)

**Comment by MurzNN** [comment]
Here is the source of Loki log lines in JSON format, from which I try to display the Graph: 
```
2021-01-01 01:01:00 {"person_names": {"Alice": 123, "Bob": 456}}
2021-01-01 01:02:00 {"person_names": {"Bob": 23, "John": 12}}
2021-01-01 01:03:00 {"person_names": {"John": 52}}
2021-01-01 01:04:00 {"person_names": {"Alice": 123, "Bob": 53, "Tom": 22}}
```
So I converted it to labels, excluding other values (because real json contains much more data) using LogQL like this:
```
{job="my_app"} | json data="person_names" | line_format `{"label":{{.data}}}` | json | __error__ = ""
```
and to table using Grafana Transformation "Labels to fields" + "Filter by name".
But can't do further conversion to timeline graph with separate line per each person...

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!

**Comment by MurzNN** [comment]
Issue https://github.com/grafana/grafana/issues/82497
